lib/install/management/appliance.go

```diff
@@ -660,6 +739,132 @@ func (d *Dispatcher) createAppliance(conf *config.VirtualContainerHostConfigSpec
 	return nil
 }
 
+// createVCHInventoryFolders creates the default structure of the inventory for the VCH and returns the object reference to the last folder created
+func (d *Dispatcher) createVCHInventoryFolders(spec *types.VirtualMachineConfigSpec) (*object.Folder, error) {
+	d.op.Info("Creating VCH inventory folders")
+	vchName := spec.Name
+	clusterName := path.Base(d.session.ClusterPath)
+	consolidated := clusterName + "/" + strings.TrimLeft(d.vchPoolPath, d.session.ClusterPath)
+
+	// assemble creation list
+	folders := strings.Split(consolidated, "/")
+	folders = append(folders, vchName)
+	d.op.Debugf("Determined inventory targets for creation: %s", folders)
+
+	// now that we have the list of needed constructions we need to assemble it.
+	existingFolder := d.session.VMFolder
+	for _, fname := range folders {
```

Comment:
Find all the targets that you can first. Then create the rest in order to avoid needless govmomi calls.

Reply:
This eliminates the need for us to determine if a folder already exists or not when advancing the pointer.